scripts/results_generation/results_gmcc_metric.py

```diff
@@ -0,0 +1,254 @@
+import os
+
+import json
+import numpy as np
+import pandas as pd
+
+from team.utility.accuracy_metric import symmetric_gmcc
+
+
+def extract_data_into_numpy(data_path: str, keyword: str) -> np.ndarray:
+    """
+    Extract data from json dictionary into numpy array
+
+    :param data_path: data path
+    :param keyword: string to select tcp or joint data
+    :return: extracted data in numpy array
+    """
+    if keyword == "tcp":
+        list_keys = ["tcp_x", "tcp_y", "tcp_z"]
+    elif keyword == "joint":
+        list_keys = ["joint_1", "joint_2", "joint_3", "joint_4", "joint_5", "joint_6"]
+    else:
+        raise ValueError("Keywork argument is not valid, insert 'tcp' or 'joint'")
+    df = pd.read_json(data_path)
+    data = df[list_keys]
+    return data.to_numpy()
+
+
+def get_trajectory_data(data_path: str) -> np.ndarray:
+    """
+    Extract trajectory data from json dictionary
+
+    :param data_path: data path
+    :return: trajectory
+    """
+    with open(data_path, 'r') as f:
+        data = json.load(f)
+    traj = data["trajectory"]
+    return np.array(traj)
+
+
+def first_is_longer(traj_1: np.ndarray, traj_2: np.ndarray) -> bool:
+    """
+    Assess if first trajectory is longer than second one
+
+    :param traj_1: first trajectory to compare
+    :param traj_2: second trajectory to compare
+    :return: if first trajectory is longer than second one
+    """
+    return np.shape(traj_1)[0] > np.shape(traj_2)[0]
+
+
+def get_motion_onset_idx(trajectory: np.ndarray, tol: float) -> int:
+    """
+    Get index at which motion onset occurs
+
+    :param trajectory: data
+    :param tol: tolerance to assess motion onset
+    :return: index at which motion starts
+    """
+    nb_samples = np.shape(trajectory)[0]
+    dims = np.shape(trajectory)[1]
+    list_idx = []
+    # Loop on each dimension
+    for i in range(dims):
+        j = 0
+        while (abs(trajectory[j, i] - trajectory[j+1, i]) < tol):
+            j = j+1
+            if j == nb_samples - 1:
+                break
+        list_idx.append(j)
+    # Take lowest index as some dimensions might not be affected by motion
+    return min(list_idx)
+
+
+def pad_end(trajectory: np.ndarray, final_len: int) -> np.ndarray:
+    """
+    Pad end of trajectory
+
+    :param trajectory: data
+    :param final_len: final trajectory length
+    :return: padded trajectory
+    """
+    nb_samples = final_len - np.shape(trajectory)[0]
+    nb_dims = np.shape(trajectory)[1]
+    extension = np.full_like(trajectory, trajectory[-1, :], shape=(nb_samples, nb_dims))
+    return np.row_stack([trajectory, extension])
+
+
+def comparison_setup(regression: np.ndarray, reproduction: np.ndarray) -> list[
+        np.ndarray]:
+    """
+    Manipulate regression and reproduction trajectories to bring them at same length
+    1. Trim initial static portion of both trajectories, not moving data
+    2. Check which trajectory is longer
+    3. Pad shorter trajectory to same length
```

Comment:
I think I understand why it's needed but I thought that GMCC took care of scaling effects? Do you need to do this preprocessing? At the very least, it seems weird to have to add padding :).